modules/sfp_snov.py

```diff
@@ -0,0 +1,199 @@
+        return ["DOMAIN_NAME", "INTERNET_NAME"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["EMAILADDR"]
+    
+    # Get Authentication token from Snov.IO API
+    def queryAccessToken(self):
+        params = {
+            'grant_type': "client_credentials",
+            'client_id': self.opts['client_id'],
+            'client_secret': self.opts['client_secret']
+        }
+
+        headers = {
+            'Accept': "application/json",
+        }
+        
+        res = self.sf.fetchUrl(
+            'https://api.snov.io/v1/oauth/access_token?' + urllib.parse.urlencode(params),
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        if not res['code'] == '200':
+            self.sf.error("Could not authenticate credentials", False)
+            self.errorState = True 
+            return None
+        try:
+            # Extract access token from response
+            content = res.get('content')
+            if isinstance(content, str):
+                accessToken = json.loads(content).get('access_token')
+            else:
+                accessToken = res.get('access_token')
+
+            return str(accessToken)
+        except Exception: 
+            self.sf.debug("Could not fetch access token")
+            self.errorState = True
+            return None
+        
+    # Fetch email addresses related to target domain
+    def queryDomainName(self, qry, accessToken, currentOffset):
+        params = {
+            'domain': qry.encode('raw_unicode_escape').decode("ascii", errors='replace'),
+            'access_token': accessToken,
+            'type': "all",
+            'limit': str(self.limit),
+            'offset': str(currentOffset)
+        }
+
+        headers = {
+            'Accept' : "application/json",
+        }
+
+        res = self.sf.fetchUrl(
+            'https://api.snov.io/v1/get-domain-emails-with-info',
+            postData=params,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        if not res['code'] == '200':
+            self.sf.debug("Could not fetch email addresses")
+            return None
+
+        return res.get('content')
+
+
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Always check if the API key is set and complain if it isn't, then set
+        # self.errorState to avoid this being a continual complaint during the scan.
+        if self.opts['client_id'] == "" or self.opts['client_secret'] == "":
+            self.sf.error("You enabled sfp_snov but did not set an client_id and/or client_secret", False)
```

Comment:
"set an" -> "set a", and replace client_id and/or client_secret with "Client ID and/or Client Secret" so that the user can identify the configuration settings they need to change.